packages/core/core/src/Transformation.js

```diff
@@ -339,7 +342,11 @@ export default class Transformation {
     configs.set('parcel', config);
 
     for (let [moduleName] of config.devDeps) {
-      let plugin = await parcelConfig.loadPlugin(moduleName);
+      let plugin = await parcelConfig.loadPlugin({
+        packageName: moduleName,
+        resolveFrom: parcelConfig.filePath,
```

Comment:
same

Reply:
My suggestion to fix this (and the [comment above](https://github.com/parcel-bundler/parcel/pull/4070/files#r374478085)) would be to update `parcelConfig.getTransformers()` to return `Array<{name, plugin, resolvedFrom}>` (not sure that's the best name but you get it). Then we could get rid of this for loop and add a new for loop between 366 and 367 that would look like this:

```
for (let {name, resolvedFrom} of transformers) {
  let thirdPartyConfig = await this.loadTransformerConfig({
    filePath,
    plugin: name,
    parcelConfigPath: resolvedFrom,
    isSource,
  });

  configs.set(name, thirdPartyConfig);
}
```

Note: `ConfigLoader` would need to be updated to conditionally call the plugin's `loadConfig` if it exists, which is probably where that check belongs anyway.

Then the correct config file path will be attached to the meta on the configRequest, which can be used up above where we `loadPlugin` to preserialize.

Reply:
I'm not sure I completely understand, than why were plugins being loaded twice before?

Reply:
> Note: ConfigLoader would need to be updated to conditionally call the plugin's loadConfig if it exists, which is probably where that check belongs anyway.

I think this is already the case?

Reply:
> I'm not sure I completely understand, than why were plugins being loaded twice before?

Short answer is I don't know if there was a good reason, but it seemed harmless because plugin loading was cached.

> I think this is already the case?

Ah, yeah you're right.

The changes you made look good to me 👍

Reply:
Awesome thanks for clearing that up and having another look at it